app/models/profile.rb

```diff
@@ -197,11 +197,8 @@ def deactivate_due_to_gpo_expiration
   end
 
   def deactivate_due_to_in_person_verification_cancelled
-    update!(
-      active: false,
-      deactivation_reason: :verification_cancelled,
-      in_person_verification_pending_at: nil,
-    )
+    update!(active: false, in_person_verification_pending_at: nil)
+    update!(deactivation_reason: :verification_cancelled) if !deactivation_reason.present?
```

Comment:
Is there a way to combine this into one UPDATE to avoid the extra database call?

Reply:
maybe like this?
```suggestion
    to_update = {
      active: false,
      in_person_verification_pending_at: nil
    }.tap do |attrs|
      attrs[:deactivation_reason] = :verification_cancelled if !deactivation_reason.present?
    end
    update!(to_update)
```

Reply:
or maybe something chaotic like this:
```suggestion
    update!(
      active: false,
      in_person_verification_pending_at: nil,
      deactivation_reason: deactivation_reason.presence || :verification_cancelled,
    )
```

Reply:
I had never seen tap before- helpful method. Ultimately, I went went suggestion 2. Resolved with commit d0b5cb4c3572c0b8f4b59fd345c1e6468f32a6ec